### PR TITLE
Update WCPay promo requirements and ensure dismiss

### LINKED ISSUE #4834: Update Payments in WC Core requirements and ensure it's dismissed on every scenario
### Describe the bug
According to this conversation parYuf-sD-p2#comment-779 we need to dismiss the Payments menu that is created in the WC Core incentive offer screen when the user selects to not install WC Payments, no matter if the survey is completed.

And update offer requirements to only show it when there is another payment gateway already installed.

### PR DESCRIPTION
**Pull request by ismaeldcom**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:
Closes https://github.com/Automattic/woocommerce-payments/issues/4834

- Update the promotion requirements to only show it for merchants with an additional payment gateway installed, using as source a static list of US gateways IDs from WC store.
- Ensure the promotion is dismissed after clicking on **No thanks**, even on modal close.

### How to test the changes in this Pull Request:
It's easier to disable the unneeded conditions on [`WcPayWelcomePage.php`](https://github.com/woocommerce/woocommerce/blob/6b80bbd1aca1e3031bd1f8693eb00b3ca4e707cf/plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php) to meet the requirements.

#### New requirement
1. The promo menu should be hidden without any of [the payment gateways](https://woocommerce.com/product-category/woocommerce-extensions/payment-gateways/?categoryIds=1023&collections=product&page=1&country=United+States) enabled.
2. Enabling any of them on **WooCommerce → Settings → Payments** should display the promo menu.

ℹ️  You can use `Direct bank transfer` gateway, included in WC, as an example to meet the requirement.

#### Promotion dismiss
1. Having the promo **Payments** menu visible, click on it.
2. Click on **No thanks** and test that the following actions hide the promotion menu.
   - Click on the modal **x** button.
   - Click on the first button.
   - Click on the second button.

ℹ️ To restore the promotion you'll need to remove or update `wc_calypso_bridge_payments_dismissed` option.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

### FOR PR REVIEWER ONLY:

-   [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php [hunk 1]
```diff
@@ -146,4 +152,28 @@ private function is_user_in_treatment_mode() {
 		return $abtest->get_variation( self::EXPERIMENT_NAME ) === 'treatment';
 	}
 
+	/**
+	 * Checks if there is another payment gateway installed using PaymentGatewaySuggestions.
+	 *
+	 * @return bool Whether there is another payment gateway installed.
+	 */
+	private function is_another_payment_gateway_installed() {
```
Comment (chihsuan) [inline plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php:192]: Hey @ismaeldcom Can we use `WC()->payment_gateways->payment_gateways()` instead? I think it returns the installed payment gateways.

Comment (ismaeldcom) [reply plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php:192]: Thanks for the suggestion @chihsuan! It is what I used at first with `WC()->payment_gateways()->get_payment_gateway_ids()`. Unfortunately, there are some gateways that where the ID doesn't match with the `PaymentGatewaySuggestions` ones.

That's why I ended up comparing the plugin slug from `get_plugins()` and the `plugins` prop on `PaymentGatewaySuggestions`.

I've should mention it on the PR description 🙂 

Another solution could be adding a static list of gateways ID to check against the active ones. Although reusing the suggestion sounds simpler.

Comment (ismaeldcom) [reply plugins/woocommerce/src/Internal/Admin/WcPayWelcomePage.php:192]: I've finally added a static list in dce3723 to have better control on which gateways will be required to meet the promo requirements. As for now we want to check for any other US based gateway on WC store.

**Review by oaratovskyi — APPROVED** [review]
Thanks for taking care of that @ismaeldcom !
I've tested by commenting next conditions locally:
- wcpay installed
- 90 days
- experiment

After installing Stripe plugin and setting it up I was able to see the page.

Clicked X, Just remove and remove and give feedback - everything works! Liked the static list of plugins - neat and straightforward

LGTM!
:shipit:

**Review by moon0326 — APPROVED** [review]
LGTM and tested well for me!

🚀

**Pull request merged**